### DISCUSSION TITLE
Slurm script fails when I request mpirun/ntasks

### DISCUSSION THREAD

**Discussion by slabhan1** [NASA-LIS/LISF#1641] [Q&A]
I am trying to run a LIS (module openmpi_5.0.0) with the following slurm script on Milan: 
```
#!/bin/bash

#SBATCH -J cold_30node
#SBATCH --constraint=mil 
#SBATCH --ntasks=1

#SBATCH --time=00:00:40
#SBATCH --output=./LDAS_irri/lisjob.out
#SBATCH -D ./wd/
#SBATCH --no-requeue
#SBATCH --mail-type=ALL
#SBATCH --error=./LDAS_irri/error.log
module purge
source $HOME/.profile


module use --append $HOME/privatemodules

module load lisf_7.5_intel_2023.2.1_openmpi_5.0.0

time mpirun -np 1 ./LIS -f ./LDAS_irri/lis_config_2001

exit 0
```

Which throws the error: 
![Screenshot 2024-11-18 at 6 33 57 PM (2)](https://github.com/user-attachments/assets/47575bba-a9c3-4cf7-bcf2-4c20e0e9f6a9)

I have tried to run this without "mpirun -np 1" and "ntasks = 1" and it works just fine, so I imagine it's not an issue with the executable/config. I can't figure out why I can't run this with mpirun, please help!  
mpirun --version
mpirun (Open MPI) 5.0.0

sinfo --version
slurm 22.05.11

**Comment by dmocko** [comment]
Hi @slabhan1 

Sorry for the delay in responding.  If you are only running with 1 task, you can just run with this command:

`./LIS -f ./LDAS_irri/lis_config_2001`

It sounds like this is already working for you.  I think "mpirun" can cause errors when running with just 1 task.

If that's not the issue, are you sure that you have both compiled and submitted the job from a Milan node?

I see you are on Discover.  You can point me to your LISF/lis/make/configure.lis (ensure permissions are open) and I'll take a look.

I just posted a separate discussion.  Please see if Issue 5 in it might apply to you.  https://github.com/NASA-LIS/LISF/discussions/1642

My apologies, I see that we never responded to your question from March:  https://github.com/NASA-LIS/LISF/discussions/1503

Would you still like us to respond to that other question?

**Comment by slabhan1** [comment]
Hi David,
Thank you, those were helpful leads, and solved the issue. I don't think I need help to use ERA5 anymore, I have been using MERRA2.
Both these discussions can be closed! Thank you!
________________________________
